libdevcore/CMakeLists.txt

```diff
@@ -34,6 +34,6 @@ set(sources
 )
 
 add_library(devcore ${sources})
-target_link_libraries(devcore PUBLIC jsoncpp Boost::boost Boost::filesystem Boost::regex Boost::system)
```

Comment:
Is explicitly linking against `std::filesystem` really necessary?

Reply:
As far as I know it is needed depending on the compiler version - but we need to be a bit careful that this actually works with all versions of clang and gcc like this - hopefully cmake is clever enough to always define the target correctly, but I haven't verified that.

Reply:
Judging from both the build failures and the fact that I didn't find cmake documentation about the ``std::filesystem`` target, it might not even exist, though :-).

I think we'll need to link against ``stdc++fs`` before gcc 9... not sure about clang... and we have to be careful about https://bugs.launchpad.net/ubuntu/+source/gcc-8/+bug/1824721, though - but it sounds like that's fixed by now.

@chfast You had this issue at some point recently, if I'm not mistaken, right? What was your solution in the end? Do you happen to have a nice cmake snippet for deciding whether to link against ``stdc++fs``?

Reply:
I think the status is like that:
- You have to link with `stdc++fs` in GCC 8,
- You don't in GCC 9,
- It is not supported in GCC 7-,
- On macos where libc++ is used it `std::filesystem` was not supported. I had to use third party library.
- Visual Studio 2017 is fine.

There are my workarounds:
- https://github.com/ethereum/evmone/blob/master/test/bench/CMakeLists.txt#L17-L36
- https://github.com/ethereum/evmone/blob/master/test/bench/bench.cpp#L20-L29

Reply:
Ok, thanks - what a mess :-). Especially the macos part... I'm reading that nowadays there is a libc++fs library as well, but maybe that's too recent to be used so far...

Reply:
If there is still no ``std::filesystem`` support for macos, we could in solidity probably just fall back to ``boost::filesystem`` there for now...

Reply:
I would say let's use boost::filesystem throughout until there is support on macos.

Reply:
This describes the std::filesystem support in libc++: https://libcxx.llvm.org/docs/UsingLibcxx.html#using-filesystem

For macos I'm restricted by XCode version available on CI. IIRC, the XCode version of clang toolchain do not provide experimental parts. But I have to recheck this, maybe in the newest version it works now.

Reply:
Ok, it looks the `std::filesystem` is available in XCode 11 (currently beta) which has clang ~8 (Apple do not publish what version of vanilla clang their compiler is based on). But they are still quite conservative - you can include `std::filesystem` but not use it unless you build for macos 10.15: https://circleci.com/gh/ethereum/evmone/3184

Reply:
Hm... that's a bit silly on their part, but what can one do...